portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,44 +45,135 @@
 public class DataServlet extends HttpServlet {
 
   private final ArrayList<UserComment> userComments = new ArrayList<>();
+  private final String defaultMaxComment = "20";
 
   /*
    * Called when a client submits a GET request to the /data URL
    * Displays all recorded user comments on page
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int maxCommentDisplay = Integer.parseInt(getFieldFromResponse(request, "maxcomments", defaultMaxComment));
+    Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
```

Comment:
Sorting date-time string is problematic, as you already observed :)

I think you could store the timestamp instead, and only convert it to the date-time string in the client.